src/main/java/org/jenkinsci/plugins/workflow/cps/steps/SetStatusStep.java

```diff
@@ -0,0 +1,69 @@
+package org.jenkinsci.plugins.workflow.cps.steps;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import hudson.Extension;
+import hudson.model.Result;
+import org.jenkinsci.plugins.workflow.actions.FlowNodeStatusAction;
+import org.jenkinsci.plugins.workflow.graph.FlowNode;
+import org.jenkinsci.plugins.workflow.steps.Step;
+import org.jenkinsci.plugins.workflow.steps.StepContext;
+import org.jenkinsci.plugins.workflow.steps.StepDescriptor;
+import org.jenkinsci.plugins.workflow.steps.StepExecution;
+import org.jenkinsci.plugins.workflow.steps.SynchronousStepExecution;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import javax.annotation.Nonnull;
+import java.util.Collections;
+import java.util.Set;
+
+public final class SetStatusStep extends Step {
```

Comment:
Yeah, I wanted something like this but don't think it'll work as implemented due to the issue above with status within a block -- I think it needs to touch some sort of common status/Result within the running block.